antarest/storage/service.py

```diff
@@ -92,97 +105,116 @@ def export_study(
+        uuid, _ = StorageServiceUtils.extract_info_from_url(path)
+        md = self._get_metadata_according_permission(params.user, uuid)
+        self.importer_service.upload_matrix(md, path, data)
 
     def import_study(
         self, stream: IO[bytes], params: RequestParameters
     ) -> str:
-        uuid = self.importer_service.import_study(stream)
+        md = Metadata(id=str(uuid4()), workspace="default")
+        md = self.importer_service.import_study(md, stream)
         status = (
             StudyContentStatus.ERROR
-            if self.study_service.check_errors(uuid)
+            if self.study_service.check_errors(md)
             else StudyContentStatus.VALID
         )
-        self._save_metadata(uuid, params.user, content_status=status)
-        return uuid
+        self._save_metadata(md, owner=params.user, content_status=status)
+        return str(md.id)
 
     def import_output(
         self, uuid: str, stream: IO[bytes], params: RequestParameters
     ) -> JSON:
-        self._check_user_permission(params.user, uuid)
-        res = self.importer_service.import_output(uuid, stream)
+        md = self._get_metadata_according_permission(params.user, uuid)
+        res = self.importer_service.import_output(md, stream)
         return res
 
     def edit_study(
         self, route: str, new: JSON, params: RequestParameters
     ) -> JSON:
-        uuid, _, _ = self.study_service.extract_info_from_url(route)
-        self._check_user_permission(params.user, uuid)
-        return self.study_service.edit_study(route, new)
+        uuid, url = StorageServiceUtils.extract_info_from_url(route)
+        md = self._get_metadata_according_permission(params.user, uuid)
+        return self.study_service.edit_study(md, url, new)
 
     def _save_metadata(
         self,
-        uuid: str,
-        user: Optional[User],
+        metadata: Metadata,
+        owner: Optional[User] = None,
         content_status: StudyContentStatus = StudyContentStatus.VALID,
+        group: Optional[Group] = None,
     ) -> None:
-        if not user:
+        if not owner and not group:
             raise UserHasNotPermissionError
 
-        info = self.study_service.get_study_information(uuid)["antares"]
-        meta = Metadata(
-            id=uuid,
-            name=info["caption"],
-            version=info["version"],
-            author=info["author"],
-            created_at=datetime.fromtimestamp(info["created"]),
-            updated_at=datetime.fromtimestamp(info["lastsave"]),
-            content_status=content_status,
-            users=[user],
-        )
-        self.repository.save(meta)
+        info = self.study_service.get_study_information(metadata)["antares"]
+
+        metadata.name = info["caption"]
+        metadata.version = info["version"]
+        metadata.author = info["author"]
+        metadata.created_at = datetime.fromtimestamp(info["created"])
+        metadata.updated_at = datetime.fromtimestamp(info["lastsave"])
```

Comment:
it should be the current datetime. and the underlying file should be updated. (i will create an issue for it)